catkin_ws/src/arena_camera/src/arena_camera_node.cpp

```diff
@@ -1269,13 +1269,16 @@ void ArenaCameraNode::spin()
     }
 
     if ((getNumSubscribersRect() > 0 || iox_state_->iox_rect.hasSubscribers()) &&
-      camera_info_manager_->isCalibrated())
+      camera_info_manager_->isCalibrated() && pinhole_model_->initialized())
     {
-      cv_bridge_img_rect_->header.stamp = img_raw_msg_.header.stamp;
-      assert(pinhole_model_->initialized());
-      cv_bridge::CvImagePtr cv_img_raw = cv_bridge::toCvCopy(img_raw_msg_, img_raw_msg_.encoding);
       pinhole_model_->fromCameraInfo(camera_info_manager_->getCameraInfo());
-      pinhole_model_->rectifyImage(cv_img_raw->image, cv_bridge_img_rect_->image);
+
+      const cv::Mat img_raw = cv_bridge::toCvCopy(img_raw_msg_)->image;
+      cv::Mat img_rect{ };
+
+      pinhole_model_->rectifyImage(img_raw, img_rect);
+
+      cv_bridge_img_rect_ = new cv_bridge::CvImage(img_raw_msg_.header, img_raw_msg_.encoding, img_rect);
```

Comment:
didn't look for myself, but are all fields getting set correctly in this call?
uint32 height
uint32 width
uint8 is_bigendian
uint32 step

Reply:
yes, they are set